lucille-core/src/test/java/com/kmwllc/lucille/connector/jdbc/DatabaseConnectorTest.java

```diff
@@ -249,4 +249,74 @@ public void testClose() throws ConnectorException, SQLException {
     assertTrue(connector.isClosed());
     assertEquals(1, dbHelper.checkNumConnections());
   }
+
+  @Test
+  public void testReservedFieldError() throws ConnectorException, SQLException {
+    HashMap<String, Object> configValues = new HashMap<>();
+    configValues.put("name", connectorName);
+    configValues.put("pipeline", pipelineName);
+    configValues.put("driver", "org.h2.Driver");
+    configValues.put("connectionString", "jdbc:h2:mem:test");
+    configValues.put("jdbcUser", "");
+    configValues.put("jdbcPassword", "");
+    configValues.put("sql", "select * from table_with_id_column");
+    configValues.put("idField", "other_id");
+
+    Config config = ConfigFactory.parseMap(configValues);
+    DatabaseConnector connector = new DatabaseConnector(config);
+
+    try {
+      connector.execute(publisher);
+    } catch (ConnectorException e) {
+      // expected
+      assertEquals("Field name \"id\" is reserved, please rename it or add it to the ignore list",

```

Comment:
This assertion won't be tested if no exception is thrown.
Better to use assertThrows here. See https://junit.org/junit4/javadoc/latest/org/junit/Assert.html#assertThrows(java.lang.Class,%20org.junit.function.ThrowingRunnable)

Reply:
I like your suggestion. I think initially I was going to add a fail statement like this
```java
    try {
      connector.execute(publisher);
      fail("Should have thrown an exception");
    } catch (ConnectorException e) {
      // expected
      assertEquals("Field name \"id\" is reserved, please rename it or add it to the ignore list",
          e.getCause().getMessage());
    }
```
is this method also acceptable?

Reply:
The fail() approach works, but assertThrows is the recommended way and is a little more concise. Go with assertThrows unless there's a specific reason why fail() is better in your scenario.